### PR TITLE
Add grpc status when wireguard mode is connected

### PR DESCRIPTION
**Pull request by neacsu**

### CODE REVIEW

**Review by octol — APPROVED** [review]

File: proto/nym/vpn.proto [hunk 1]
```diff
@@ -87,13 +106,10 @@ enum ConnectionStatus {
...
-  Address nym_address = 1;
-  Gateway entry_gateway = 2;
-  Gateway exit_gateway = 3;
-  Address exit_ipr = 4;
-  string ipv4 = 5;
-  string ipv6 = 6;
-  google.protobuf.Timestamp since = 7;
+  Gateway entry_gateway = 1;
+  Gateway exit_gateway = 2;
+  ConnectedStateDetails specific_details = 3;
```
Comment (doums) [inline proto/nym/vpn.proto:111]: Suggestion: `protocol_details`

File: proto/nym/vpn.proto [hunk 2]
```diff
@@ -36,6 +36,25 @@ message ExitNode {
...
+  string ipv6 = 4;
+}
+
+message WgConnectedStateDetails {
+  string entry_ipv4 = 1;
+  string exit_ipv4 = 2;
+}
+
+message ConnectedStateDetails {
+  oneof connected_state_details_enum {
```
Comment (doums) [inline proto/nym/vpn.proto:52]: Suggestion: `connected_state_details` as `_enum` prefix is redundant (yes oneof fields are converted as Rust enum)

**Review by doums — APPROVED** [review]

**Comment by octol** [comment]
Pushed a commit that resolves the compilation on linux

**Comment by neacsu** [comment]
Assigned to @zaneschepke for the necessary Kotlin changes corresponding to the uniffi changes.

**Review by rokas-ambrazevicius — APPROVED** [review]

**Comment by zaneschepke** [comment]
FYI Kotlin generation for uniffi breaks if you have enum names that match property struct names like this:

```
#[derive(uniffi::Enum, PartialEq)]
#[allow(clippy::large_enum_variant)]
pub enum NymVpnStatus {
    MixnetConnectionInfo {
        mixnet_connection_info: MixConnectionInfo,
        mixnet_exit_connection_info: MixExitConnectionInfo,
    },
    WireguardConnectionInfo {
        entry_connection_info: WireguardConnectionInfo,
        exit_connection_info: WireguardConnectionInfo,
    },
}
```

**Pull request merged**